iceoryx_utils/include/iceoryx_utils/internal/posix_wrapper/access_control.hpp

```diff
@@ -43,8 +43,12 @@ class AccessController
     /// @brief maximum number of permission entries the AccessController can store
     static constexpr int32_t MaxNumOfPermissions = 20;
 
-    /// @brief identifier for a permission entry (user, group, others, ...)
+/// @brief identifier for a permission entry (user, group, others, ...)
+#if defined(QNX) || defined(QNX__) || defined(__QNX__)
+    enum class Category : std::underlying_type(acl_tag_t)
```

Comment:
shouldn't that be `typename std::underlying_type<acl_tag_t>::type` or `std::underlying_type_t<acl_tag_t>`? At least this was necessary for clang/osx to compile. Also I am wondering if we still need that `#if/#else` or if linux can also deal with `std::underlying_type`.

Reply:
We need it. I tried it on linux and QNX and that code was compiling. It seems we need to refactor this when we work on the OSX support, though.